### PR TITLE
fix: get clientgen version from go.mod file

### PR DESCRIPTION
**Pull request by sdowell**
The version for the clientgen tools was previously hard coded, which
required a manual update and allowed for drift from the k8s version
set in the go dependencies. Parsing the version from go.mod ensures
there is no drift by default.

Based on https://github.com/GoogleContainerTools/kpt-config-sync/pull/1041

### CODE REVIEW

**Review by tiffanny29631 — APPROVED** [review]
/lgtm

want to wait for the other one to be merged?

**Comment by sdowell** [comment]
/hold

yes, the intention was to merge the other PR first. The other PR needs lgtm label

File: Makefile.gen
Comment (karlkfi) [inline Makefile.gen:30]: nit: you can grep without cat

Comment (karlkfi) [inline Makefile.gen:30]: Suggest adding the before & after versions to the commit message. Looks like v0.26.9 is the client we're using now, at least until the next bump. And the old hard coded version was v0.22.2?

Comment (sdowell) [reply Makefile.gen:30]: Yeah, the diff will be clearer once this change is rebased - but I'll update the commit message as well

Comment (sdowell) [reply Makefile.gen]: done

Comment (sdowell) [reply Makefile.gen]: done

**Review by karlkfi — APPROVED** [review]
/lgtm

**Pull request merged**
